### PR TITLE
Updates to support modelmeta/index_netcdf

### PR DESCRIPTION
**Pull request by rod-glover**
modelmeta indexing script uses several additional methods that belong in `nchelpers`.

During work on the indexing script, I discovered that the apparent intended outputs of the methods `dim_axes_from_names` and `dim_axes` had been modified undesirably in the initial version of nchelpers: Original/desired output is dict mapping dimension names to axis names, i.e., {dim: axis}. Initial version of nchelpers had output {axis: dim}.The apparent original intention is more convenient for coding the indexing script, so I restored it, updated tests for those 2 methods, and updated (internal) clients of those methods. There are no external clients at present, so this is only technically a breaking change.

@jameshiebert let's discuss if you have concerns about these changes.

Note: There are other functions in the new Python indexing script that should probably be migrated to nchelpers, but that may be subject to discussion. No need to merge this PR until discussion and modifications have been completed.

### CODE REVIEW

File: nchelpers/__init__.py [hunk 1]
```diff
@@ -253,8 +253,10 @@ def dim_names(self, var_name=None):
             return tuple(k for k in self.dimensions.keys())
 
     def dim_axes_from_names(self, dim_names=None):
-        """Translate well-known dimension names to canonical axis names.
-        Canonical axis names are 'X', 'Y', 'Z', 'T'.
+        """Map names of dimensions in file to canonical axis names, based on well-known dimension names for axes.
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: Good, thanks for adding more verbose info here.

File: nchelpers/__init__.py [hunk 2]
```diff
@@ -293,27 +297,60 @@ def dim_axes(self, dim_names=None):
...
         # Then fill in the rest from the 'axis' attributes
-        # TODO: Does this happen? i.e., when are dimension names the same as axis names?
-        # Alternatively, is this some kind of (relatively benign) programming error?
-        for axis in axis_to_dim.keys():
-            if axis in self.dimensions and axis in self.variables \
-                    and hasattr(self.variables[axis], 'axis'):
-                axis_to_dim[axis] = self.variables[axis].axis
+        for dim_name in dim_name_to_axis.keys():
+            if dim_name in self.dimensions and dim_name in self.variables \
+                    and hasattr(self.variables[dim_name], 'axis'):
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: I'm a bit concerned about conflating the conditionals of "variable exists" and "axis attribute exists". I.e. if the variable exists, but doesn't have an axis attribute, then this script will just skip it. However, I would want the script to either error out or raise a really loud warning. We should probably write an issue for this and defer.

File: nchelpers/__init__.py [hunk 3]
```diff
@@ -293,27 +297,60 @@ def dim_axes(self, dim_names=None):
...
+                    dim_name_to_axis[dim_name] = 'S'
 
-                # Apparently this is how a "space" dimension is attributed?
-                if hasattr(self.variables[axis], 'compress'):
-                    axis_to_dim[axis] = 'S'
+        return dim_name_to_axis
+
+    def axes_dim(self, dim_names=None):
+        '''Return a dictionary mapping canonical axis names to specified dimension names (or all dimensions in file).
+        ASSUMPTION: There is at most one dimension (name) per canonical axis name. If not, the mapping inversion
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: Yes, good to state this.

File: nchelpers/__init__.py [hunk 4]
```diff
@@ -500,6 +537,19 @@ def is_hydromodel_iobs_output(self):
         return self.is_hydromodel_output and self.forcing_type == 'gridded observations'
 
     @property
+    def model_type(self):
+        """String indicating what type of model the file contains.
+        Supports modelmeta/mm_cataloguer/index_netcdf.py.
+        Really rudimentary decision making about model type.
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: This is how the script was written originally, so it's fine for this PR. But we should probably log an issue against the function to improve it (i.e. cover all of the use cases that we'll be running it against).

File: tests/test_CFDataset.py [hunk 5]
```diff
@@ -181,6 +187,21 @@ def test_dim_axes2(tiny_dataset, expected):
...
+    ('downscaled', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),
+    ('hydromodel_gcm', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),  # why isn't depth in this list?
+    ('climo_gcm', {'T': 'time', 'X': 'lon', 'Y': 'lat'}),
+], indirect=['tiny_dataset'])
+def test_axes_dim(tiny_dataset, expected):
+    assert tiny_dataset.axes_dim() == expected
+
+
+# TODO: Obtain a test file with reduced dimensions and test reduced_dims with it.
+# Request sent to Stephen Sobie and Trevor Murdock 2017-06-02 for such a file.
```
Comment (jameshiebert) [inline tests/test_CFDataset.py]: Yeah, they may not know of any. I'm pretty sure that David Bronaugh would not have put compressed dimension handling in if he hadn't seen it already, so maybe some old CMIP3 files used it? I'd be surprised if Trevor or Stephen have seen (or remembered) them.

Comment (rod-glover) [inline tests/test_CFDataset.py]: Neither Trevor nor Stephen will admit to having seen/remembered one.

**Pull request merged**